pkg/cloudprovider/providers/openstack/openstack_volumes.go

```diff
@@ -201,12 +206,34 @@ func (volumes *VolumesV2) deleteVolume(volumeName string) error {
 	return err
 }
 
+// return true when volume has been attached or detached.
+// or return false.
+func (os *OpenStack) OperateHasDone(diskName string) (bool, string, error) {
+	volume, err := os.getVolume(diskName)
+	if err != nil {
+		return false, "", err
+	}
+	VolumeStatus := volume.Status
+	if (VolumeStatus == VolumeErrorStatus) {
+		glog.Errorf("volume's status is %s", VolumeStatus)
```

Comment:
Can you log the disk name when we are logging the error? Without disk name, this error log will be probably useless